lib/repl.js

```diff
@@ -1507,72 +1508,60 @@ function regexpEscape(s) {
 
 // If the error is that we've unexpectedly ended the input,
 // then let the user try to recover by adding more input.
+// Note: `e` (the original exception is not used by the current implemention,
+// but may be needed in the future.)
 function isRecoverableError(e, code) {
-  if (e && e.name === 'SyntaxError') {
-    var message = e.message;
-    if (message === 'Unterminated template literal' ||
-        message === 'Unexpected end of input') {
-      return true;
-    }
+  let recoverable = false;
+
+  // Determine if the point of the any error raised is at the end of the input.
+  // There are two cases to consider:
+  //   1.  Any error raised after we have encountered the 'eof' token.
+  //       This prevents us from declaring partial tokens (like '2e') as
+  //       recoverable.
+  //   2.  Three cases where tokens can legally span lines.  This is
+  //       template, comment, and strings with a backslash at the end of
+  //       the line, indicating a continuation.
+  acorn.plugins.repl = (parser) => {
+    parser.extend('nextToken', (nextToken) => {
+      return function(pos, message) {
+        nextToken.call(this, pos, message);
+
+        if (parser.type === acorn.tokTypes.eof) recoverable = true;
+      };
```

Comment:
`acorn.tokTypes` is commonly stored in a var called `tt` in acorn plugins.
So then it would be `tt.eof`.